tiny_dnn/core/kernels/conv2d_grad_op.h

```diff
@@ -72,8 +72,15 @@ class Conv2dGradOp : private Conv2d, public core::OpKernel {
         // initalize outputs
         fill_tensor(prev_delta, float_t(0));
         
+        // in case we want zero-pading we need to
+        // reshape the data container
+
+        if (Conv2d::params().pad_type == core::padding::same) {
+            Conv2d::copy_and_pad_input(prev_delta, prev_delta);
+        }
+
```

Comment:
@nyanp I added this fix in order to resize data container, but not sure if is okay since not pass tests